src/csharp/Microsoft.Spark/Interop/Ipc/JvmBridge.cs

```diff
@@ -267,6 +267,9 @@ private ISocketWrapper GetConnection()
                     case 'd':
                         returnValue = SerDe.ReadDouble(inputStream);
                         break;
+                    case 'm':
+                        returnValue = decimal.Parse(SerDe.ReadString(inputStream));
```

Comment:
Use decimal.Parse(SerDe.ReadString(inputStream), CultureInfo.InvariantCulture) to ensure we are using invariant culture on the wire.